downloader/src/main/kotlin/WorkingTreeCache.kt

```diff
@@ -55,11 +56,22 @@ interface WorkingTreeCache {
 
 class DefaultWorkingTreeCache : WorkingTreeCache {
     private val mutex = Mutex()
+    private val vcsPluginConfigs = mutableMapOf<String, PluginConfig>()
     private val workingTreeMutexes = mutableMapOf<String, Mutex>()
     private val workingTrees = mutableMapOf<String, WorkingTree>()
 
     private var terminated = false
 
+    /**
+     * Add [VCS plugin configurations][configs] that are applied when creating a [VersionControlSystem] instance
+     * for this working tree. Calling this function is optional.
+     */
+    fun withVcsPluginConfigs(configs: Map<String, PluginConfig>): DefaultWorkingTreeCache {
```

Comment:
To me, the semantics of "with" sound more as if *exclusively* the given `configs` would be used, but the code is actually adding to any previously added configs. So I propose to either renaming the function to `addVcsPluginConfigs()`, or to change the implementation clear `vcsPluginConfigs` before adding.